nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -386,6 +391,56 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
     plan.poses.push_back(pose);
   }
 
+
+  // Publish expansions for debug
+  if (_viz_expansions) {
+    auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
+    visualization_msgs::msg::Marker m;
+    m.header.frame_id = _global_frame;
+    m.header.stamp = _clock->now();
+    m.ns = "smach_hybrid_footprint";
+    m.type = visualization_msgs::msg::Marker::LINE_LIST;
+    m.action = visualization_msgs::msg::Marker::ADD;
+    m.scale.x = 0.05;
+    m.scale.y = 0.05;
+    m.scale.z = 0.05;
+    m.color.a = 1.0;
+    m.color.r = 0;
+    m.color.g = 0.0;
+    m.color.b = 1.0;
+    m.lifetime = rclcpp::Duration(0s);
+    m.frame_locked = false;
+
+    unsigned int id = 0;
+    for (size_t i = 0; i < plan.poses.size(); i ++)
+    {
+      const double& x = plan.poses[i].pose.position.x;
+      const double& y = plan.poses[i].pose.position.y;
+      const double& yaw = tf2::getYaw(plan.poses[i].pose.orientation);
+
+      std::vector<geometry_msgs::msg::Point> footprint;
+      transformFootprintToEdges(x, y, yaw, _footprint_spec, footprint);
+
+      m.points.clear();
+      for (auto& point : footprint) {
+        m.points.push_back(point);
+      }
+      m.id = id;
```

Comment:
why id the ID not just `i`? I don't think the separate `id` variable is required.

Reply:
Txs!